cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterResourcesTest.java

```diff
@@ -0,0 +1,137 @@
+            new ResourcesCombo(null, null, "0Gi", "2Gi", "[.spec.kafka.resources memory request must be > zero]"),
+            new ResourcesCombo(null, null, "1Gi", "0Gi", "[.spec.kafka.resources memory limit must be > zero, .spec.kafka.resources memory request must be <= limit]"),
+
+            new ResourcesCombo(null, null, null, null, null),
+            new ResourcesCombo(String.valueOf(Long.MAX_VALUE), String.valueOf(Long.MAX_VALUE), String.valueOf(Long.MAX_VALUE), String.valueOf(Long.MAX_VALUE), null),
+            new ResourcesCombo("0", "0", "0", "0", "[.spec.kafka.resources cpu request must be > zero, .spec.kafka.resources cpu limit must be > zero, "
+                    + ".spec.kafka.resources memory request must be > zero, .spec.kafka.resources memory limit must be > zero]"),
+            new ResourcesCombo("1000m", "0m", "1Gi", "0Gi", "[.spec.kafka.resources cpu limit must be > zero, .spec.kafka.resources cpu request must be <= limit, "
+                    + ".spec.kafka.resources memory limit must be > zero, .spec.kafka.resources memory request must be <= limit]"),
+            new ResourcesCombo("-1", "-1", "-1", "-1", "[.spec.kafka.resources cpu request must be > zero, .spec.kafka.resources cpu limit must be > zero, " 
+                        + ".spec.kafka.resources memory request must be > zero, .spec.kafka.resources memory limit must be > zero]")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("resourcesCombos")
+    public void withComputeResources(ResourcesCombo combo) {
+        ResourceRequirementsBuilder builder = new ResourceRequirementsBuilder();
+        if (combo.cpuRequest() != null) {
+            builder.addToRequests("cpu", new Quantity(combo.cpuRequest()));
+        }
+        if (combo.cpuLimit() != null) {
+            builder.addToLimits("cpu", new Quantity(combo.cpuLimit()));
+        }
+        if (combo.memoryRequest() != null) {
+            builder.addToRequests("memory", new Quantity(combo.memoryRequest()));
+        }
+        if (combo.memoryLimit() != null) { 
+            builder.addToLimits("memory", new Quantity(combo.memoryLimit()));
+        }
+
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editSpec()
+                    .editKafka()
+                        .withResources(builder.build())
+                    .endKafka()
+                .endSpec()
+                .build();
+
+        if (combo.error() != null) {
+            InvalidResourceException ex = assertThrows(InvalidResourceException.class, () -> KafkaCluster.fromCrd(Reconciliation.DUMMY_RECONCILIATION, kafka, VERSIONS));
+            assertThat(ex.getMessage(), equalTo(combo.error()));
+        } else {
+            KafkaCluster kc = KafkaCluster.fromCrd(Reconciliation.DUMMY_RECONCILIATION, kafka, VERSIONS);
+            assertThat(kc.getContainers(null).get(0), notNullValue());
+        }   
+    }
```

Comment:
This is more or less good ... but you are testing a static method from `ModelUtils`. You should make your life (and the life of anyone changing this in the future) and move this to `ModelUtilsTest` and test the static methods directly. That way, it will be resilient against any changes to the `KafkaCluster` class.

Reply:
I can do that, but I would still need tocreate a dedicated test class as I cannot mix `@ParameterizedTest` with `@Test`.

Reply:
Ok - I still think it makes sense to test it directly, even if it is a separate class.

Reply:
Me too.